script.js

```diff
@@ -11,29 +71,47 @@ function createListItem(taskContent) {
 
   return listItem;
 }
-
+//////////////////////////////
 if (listContainer) {
-  listContainer.addEventListener("click", handleListClick, false);
+  listContainer.addEventListener("click", handleListClick);
 }
 
 function handleListClick(event) {
   const clickedElement = event.target;
 
   if (clickedElement.tagName === "LI") {
     clickedElement.classList.toggle("checked");
+    tasks.map((task) => {
+      if (task.id === +clickedElement.id) {
+        task.status = task.status === "active" ? "completed" : "active";
+      }
+      return task;
+    });
   } else if (clickedElement.tagName === "SPAN") {
     clickedElement.parentElement.remove();
   }
 }
 
+input.addEventListener("keypress", function (event) {
+  if (event.key === "Enter") {
+    event.preventDefault();
+    addTask();
+  }
+});
+
 function addTask() {
   const taskInputValue = input.value.trim();
-
   if (taskInputValue === "") {
-    const EMPTY_INPUT_MESSAGE = "Musisz coś wpisać";
-    alert(EMPTY_INPUT_MESSAGE);
+    alert("Wprowadzono pusty tekst");
   } else {
-    const listItem = createListItem(taskInputValue);
+    const id = new Date().valueOf();
+    const listItem = createListItem(taskInputValue, id);
+
+    tasks.push({
+      id: id,
+      title: taskInputValue,
+      status: "active",
+    });
     if (listContainer) {
       listContainer.appendChild(listItem);
     }
```

Comment:
Refactor to this

```
// Modified Function: Naming and added empty string check
function addTask() {
  const taskInputValue = input.value.trim();
  
  // Added: Check for empty string
  if (!taskInputValue) {
    alert("Please enter non-empty text");
    return; // Exit function early to prevent further execution
  }

  const id = new Date().valueOf();
  const listItem = createListItem(taskInputValue, "active", id);

  tasks.push({
    id: id,
    title: taskInputValue,
    status: "active",
  });

  if (listContainer) {
    listContainer.appendChild(listItem);
  }
  input.value = "";
}

```